frontend/src/components/ToDoItem/ToDoItem.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react';
+import { ItemList } from '../../types/common';
+import './styles.css';
+
+const ToDoItem = (props: {
+  item: ItemList;
+  onUpdateItem: (item: ItemList) => void;
+  onRemoveItem: (id: string) => void;
+}) => {
+  const {
+    item: { text, id, isDone },
+    onUpdateItem,
+    onRemoveItem
+  } = props;
+  const [newValue, setNewValue] = useState(text);
+  const [isChecked, setIsChecked] = useState(isDone);
+  const [isEdit, setIsEdit] = useState(false);
+
+  const updateHandler = () => {
+    onUpdateItem({ ...props.item, text: newValue });
+    setIsEdit(false);
+  };
+
+  const editItemSwitherHandler = () => {
+    setIsEdit(true);
+  };
+
+  const editItemStatusHandler = () => {
+    setIsChecked((isChecked) => !isChecked);
+    onUpdateItem({ ...props.item, isDone: !isChecked });
+  };
+
+  return (
+    <li key={id} className="list-item">
+      <label>
+        <input type="checkbox" checked={isChecked} onChange={editItemStatusHandler} name={id} />
+        {!isEdit && <span className="label-text">{text}</span>}
+      </label>
+      {isEdit && (
+        <>
+          <input type="text" value={newValue} onChange={(e) => setNewValue(e.target.value)} />
```

Comment:
you can always show this input despite isEdit state. But change it to have disabled attribute (with CSS making it looks like casual text)

Reply:
or even better - the input is always editable. just make a focused style state

Reply:
fixed